src/lava/magma/core/process/ports/ports.py

```diff
@@ -703,12 +729,18 @@ class ReshapePort(AbstractVirtualPort):
     It is used by the compiler to map the indices of the underlying
     tensor-valued data array from the derived to the new shape."""
 
-    def __init__(self, new_shape: ty.Tuple):
+    def __init__(self,
+                 new_shape: ty.Tuple[int, ...],
+                 old_shape: ty.Tuple[int, ...]):
         AbstractPort.__init__(self, new_shape)
+        self.old_shape = old_shape
 
-    def get_transform_func(self) -> ft.partial:
+    def get_transform_func_fwd(self) -> ft.partial:
         return ft.partial(np.reshape, newshape=self.shape)
```

Comment:
Maybe add a short docstring what this does (explain fwd/bwd) and copy it to all 4 methods.

Reply:
Good point, thanks!